tool/tctl/common/plugin/entraid.go

```diff
@@ -0,0 +1,403 @@
+	p.install.entraID.cmd = parent.Command("entraid", "Install an EntraId integration.")
+	cmd := p.install.entraID.cmd
+	cmd.
+		Flag("name", "Name of the plugin resource to create").
+		Default("entra-id").
+		StringVar(&p.install.name)
+
+	cmd.
+		Flag("auth-connector-name", "Name of the SAML connector resource to create").
+		Default("entra-id-default").
+		StringVar(&p.install.entraID.authConnectorName)
+
+	cmd.
+		Flag("use-system-credentials", "Uses system credentials instead of OIDC.").
+		BoolVar(&p.install.entraID.useSystemCredentials)
+
+	cmd.Flag("default-owner", "List of Teleport users that are default owners for the imported access lists. Multiple flags allowed.").
+		Required().
+		StringsVar(&p.install.entraID.defaultOwners)
+
+	cmd.
+		Flag("access-graph", "Enables Access Graph cache build.").
+		Default("true").
+		BoolVar(&p.install.entraID.accessGraph)
+
+	cmd.
+		Flag("force", "Proceed with installation even if plugin already exists.").
+		Short('f').
+		Default("false").
+		BoolVar(&p.install.scim.force)
+}
+
+type entraSettings struct {
+	accessGraphCache *azureoidc.TAGInfoCache
+	clientID         string
+	tenantID         string
+}
+
+var (
+	errCancel = trace.BadParameter("operation canceled")
+)
+
+func (p *PluginsCommand) entraSetupGuide(proxyPublicAddr string) (entraSettings, error) {
+	fileLoc, err := pathForFile(os.Stdout, os.Stdin)
+	if err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to get file location")
+	}
+
+	buildScript, err := buildScript(proxyPublicAddr, p.install.entraID.authConnectorName, p.install.entraID.accessGraph, p.install.entraID.useSystemCredentials)
+	if err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to build script")
+	}
+
+	if err := os.WriteFile(fileLoc, []byte(buildScript), 0644); err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to write script to file")
+	}
+
+	tmpl := `Step 1: Run the Setup Script
+
+1. Open **Azure Cloud Shell** (Bash) using **Google Chrome** or **Safari** for the best compatibility.
+2. Upload the setup script using the **Upload** button in the Cloud Shell toolbar.
+3. Once uploaded, execute the script by running the following command:
+   $ bash %s
+
+**Important Considerations**:
+- You must have **Azure privileged administrator permissions** to complete the integration.
+- Ensure you're using the **Bash** environment in Cloud Shell.
+- During the script execution, you'll be prompted to run 'az login' to authenticate with Azure. **Teleport** does not store or persist your credentials.
+- **Mozilla Firefox** users may experience connectivity issues in Azure Cloud Shell; using Chrome or Safari is recommended.
+
+`
+
+	fmt.Fprintf(os.Stdout, tmpl, filepath.Base(fileLoc))
+
+	op, err := readData(os.Stdin, os.Stdout,
+		"Once the script completes, type 'continue' to proceed, 'exit' to quit",
+		func(input string) bool {
+			return input == "continue" || input == "exit"
+		}, "Invalid input. Please enter 'continue' or 'exit'.")
```

Comment:
If the user writes 'y', they'll get `Invalid input. Please enter 'continue' or 'exit'.`
But does the program exits or does it try to read again the stdin?